clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -14613,3 +14613,26 @@ void TypeRemoveRule::runRule(
 }
 
 REGISTER_RULE(TypeRemoveRule, PassKind::PK_Analysis)
+
+void CompatNVCCRule::registerMatcher(ast_matchers::MatchFinder &MF) {
+  MF.addMatcher(functionDecl().bind("FunctionDecl"), this);
```

Comment:
How about use the functionTemplateDecl as a replace of functionDecl?

Reply:
I have updated the matcher.